### PR TITLE
Fix CLI metrics commands: replace fake data with real metrics pipeline

### PR DESCRIPTION
**Pull request by ElNiak**
## Summary

- **Replace all fake/random data** in the 7 `panther metrics` CLI subcommands (`list`, `show`, `export`, `clear`, `summary`, `quick`, `backup`) with real experiment metrics loaded from disk
- **Wire metrics export into experiment lifecycle** so `MetricsCollector` data is persisted to `outputs/<timestamp>/metrics/` (JSON + CSV) during cleanup, even on failure
- **Fix multiple bugs**: broken `MetricsCollector` kwargs in `run.py`, `ResourceMonitor` import only under `TYPE_CHECKING`, missing context manager for guaranteed cleanup, hardcoded version string in `MetricsExporter`

### New files
- `panther/core/metrics/data_loader.py` — `MetricsDataLoader` utility for discovering and loading persisted metrics from experiment output directories
- `tests/unit/test_core/test_metrics_data_loader.py` — 29 unit tests for data loader
- `tests/unit/test_core/test_metrics_export_lifecycle.py` — 6 roundtrip integration tests (collect -> export -> load)

### Modified files
- `panther/cli_click/commands/metrics.py` — Rewritten: all 7 subcommands use `MetricsDataLoader` instead of `random.uniform()`/hardcoded values
- `panther/cli_click/commands/run.py` — Fixed `MetricsCollector` init kwargs; wrapped `ExperimentManager` in context manager for guaranteed cleanup
- `panther/core/experiment_manager.py` — Added metrics export (JSON + CSV) in `cleanup()`
- `panther/core/metrics/__init__.py` — Added `MetricsDataLoader` to exports
- `panther/core/metrics/metrics_exporter.py` — Replaced hardcoded `"1.0.0"` with dynamic version
- `panther/core/observer/impl/metrics_observer.py` — Moved `ResourceMonitor` import from `TYPE_CHECKING` to runtime
- `tests/cli_click/unit/commands/test_all_commands.py` — Updated metrics command tests to mock `MetricsDataLoader`

## Test plan

- [x] 110/110 unit tests pass (`pytest tests/ -v`)
- [x] CLI smoke test with no experiment data — all commands show "No metrics data available" warning
- [x] End-to-end: ran `experiment_config_example_minimal_docker.yaml` with `--enable-metrics`, verified metrics files created in `outputs/*/metrics/`, then verified all CLI commands read real data
- [x] Metrics coherence validated: timing sums match experiment duration, CPU/memory values realistic, timestamps monotonic, CSV matches JSON

## Summary by Sourcery

Replace the CLI metrics commands to read and operate on real experiment metrics persisted to disk, and wire metrics collection/export into the experiment lifecycle so metrics are reliably saved and consumable.

New Features:
- Introduce MetricsDataLoader to discover, load, and summarize persisted experiment metrics for CLI consumption.
- Add support in the metrics CLI commands to select experiments and output roots via shared --experiment-dir/--output-dir options.

Bug Fixes:
- Correct MetricsCollector initialization and resource monitoring in the run command, and ensure ExperimentManager.cleanup (including metrics export) always runs via a context manager.
- Fix metrics observer to import ResourceMonitor at runtime instead of only during type checking.
- Replace the hardcoded panther_version in MetricsExporter JSON output with a dynamically resolved package version.

Enhancements:
- Rewrite all panther metrics subcommands (list, show, export, clear, summary, quick, backup) to use MetricsDataLoader and real exported metrics instead of mock data.
- Extend ExperimentManager.cleanup to export collected metrics as JSON and CSV files under each experiment's metrics/ directory.
- Refine metrics CLI UX with clearer messaging when no metrics are found and more informative summaries and exports.

Tests:
- Add comprehensive unit tests for MetricsDataLoader covering discovery, loading, metric enumeration, value retrieval, and summary generation.
- Add integration tests covering the full metrics lifecycle from MetricsCollector through MetricsExporter to MetricsDataLoader round-trips.
- Update CLI tests for metrics commands to exercise the new real-data behavior and options, and to drop obsolete subcommands/report-format tests.

### CODE REVIEW

File: panther/cli_click/commands/metrics.py [hunk 1]
```diff
@@ -209,261 +216,226 @@ def show(ctx, metric, limit):
...
+    elif fmt == "csv":
+        # Flatten timing metrics + raw counters/gauges into rows
+        rows = []
+        for name, value in data.get("timing_metrics", {}).items():
+            rows.append({"metric": name, "value": value, "type": "timing"})
+        raw = data.get("raw_metrics", {})
+        if isinstance(raw, dict):
+            for section in ("counters", "gauges"):
+                for name, value in raw.get(section, {}).items():
+                    rows.append({"metric": name, "value": value, "type": section})
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply panther/cli_click/commands/metrics.py]: Addressed: CSV export in `metrics.py:259-309` now includes ALL categories: timing, raw counters/gauges/histograms, resource_metrics, phase_metrics, and error_metrics. The export is comprehensive.

File: tests/cli_click/unit/commands/test_all_commands.py [hunk 2]
```diff
@@ -265,102 +265,150 @@ def test_metrics_help(self, cli_runner):
...
 
-    def test_metrics_list(self, cli_runner):
-        """Test listing available metrics."""
-        result = cli_runner.invoke(cli, ["metrics", "list"])
+    def test_metrics_list_no_data(self, cli_runner, temp_dir):
+        """Test listing metrics when no data is available."""
+        result = cli_runner.invoke(
+            cli, ["metrics", "list", "--output-dir", str(temp_dir)]
+        )
         assert result.exit_code is not None
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply tests/cli_click/unit/commands/test_all_commands.py]: Addressed: Tests now assert `exit_code == 0` and verify output content. E.g. `test_metrics_list_with_data` asserts 'test_dur' in output; `test_metrics_list_filter_no_match` asserts 'No metrics found matching' in output; `test_metrics_summary_with_data` checks formatted values like '42.50s' and '55.0%'.

[the remark this answers is not in the document]
Comment (ElNiak) [reply tests/cli_click/unit/commands/test_all_commands.py]: Addressed: `test_metrics_export_with_data` now verifies file exists, is non-empty, and parses as valid JSON with expected keys (`timing_metrics`, `test_dur`).

[the remark this answers is not in the document]
Comment (ElNiak) [reply tests/cli_click/unit/commands/test_all_commands.py]: Addressed: Both positive and negative filter cases covered. `test_metrics_list_with_filter` (line 375) asserts `cpu_duration` in output; `test_metrics_list_filter_no_match` (line 577) asserts no matching metrics and 'No metrics found matching' warning.

[the remark this answers is not in the document]
Comment (ElNiak) [reply tests/cli_click/unit/commands/test_all_commands.py]: Addressed: Both edge cases covered. `test_metrics_with_experiment_dir_no_metrics` (line 610) tests a bare directory and asserts "No metrics data found" in output. `test_metrics_list_multiple_experiments_selects_latest` (line 623) creates two experiments with different mtimes and verifies the latest is selected. Resolving.

File: tests/cli_click/unit/commands/test_all_commands.py [hunk 3]
```diff
@@ -513,8 +561,8 @@ def test_check_argument_validation(self, cli_runner):
 
     def test_metrics_argument_validation(self, cli_runner):
         """Test metrics command argument validation."""
-        # Invalid format
-        result = cli_runner.invoke(cli, ["metrics", "report", "--format", "invalid"])
+        # Invalid export format
+        result = cli_runner.invoke(cli, ["metrics", "export", "--format", "invalid"])
         assert result.exit_code != 0 or result.exit_code is None
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply tests/cli_click/unit/commands/test_all_commands.py]: Addressed: `test_metrics_argument_validation` (line 814) asserts `exit_code \!= 0` and verifies `"invalid" in result.output.lower()` for unsupported format values. Resolving.

File: tests/unit/test_core/test_metrics_data_loader.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+class TestFindExperimentsWithMetrics:
+    def test_finds_experiments_with_metrics(self, experiment_with_metrics):
+        root, exp_dir = experiment_with_metrics
+        loader = MetricsDataLoader(output_dir=root)
+        results = loader.find_experiments_with_metrics()
+        assert len(results) == 1
+        assert results[0] == exp_dir
+
+    def test_excludes_experiments_without_metrics(self, experiment_without_metrics):
+        root, _ = experiment_without_metrics
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply tests/unit/test_core/test_metrics_data_loader.py:129]: Addressed: `test_returns_experiments_ordered_by_mtime_descending` in test_metrics_data_loader.py covers ordering of `find_experiments_with_metrics()`. Additionally, `test_load_metrics_selects_newest_experiment` (newly added) covers `load_metrics()` auto-selecting newest experiment by mtime. CLI-level coverage via `test_metrics_list_multiple_experiments_selects_latest`. Resolving.

File: tests/unit/test_core/test_metrics_data_loader.py [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+class TestLoadMetrics:
+    def test_loads_valid_json(self, experiment_with_metrics, sample_metrics_json):
+        root, exp_dir = experiment_with_metrics
+        loader = MetricsDataLoader(output_dir=root)
+        data, path = loader.load_metrics(experiment_dir=exp_dir)
+        assert data is not None
+        assert data["export_metadata"]["panther_version"] == "1.0.0"
+        assert path == exp_dir
+
+    def test_auto_discovers_latest(self, experiment_with_metrics):
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply tests/unit/test_core/test_metrics_data_loader.py:169]: Addressed: `test_load_metrics_returns_none_when_latest_is_corrupt` (line 222 in test_metrics_data_loader.py) explicitly tests this scenario: creates an older valid experiment and a newer corrupt one, verifies `load_metrics()` returns `data=None, path=newer_exp`. Comment in test documents: "The loader picks the latest experiment and returns None for corrupt data without falling back to older experiments." This is intentional design. Resolving.

File: tests/unit/test_core/test_metrics_export_lifecycle.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        # Load back with DataLoader
+        loader = MetricsDataLoader(output_dir=tmp_path)
+        data, exp_path = loader.load_metrics(experiment_dir=tmp_path / "experiment")
+        assert data is not None
+
+        # Verify structure
+        assert "export_metadata" in data
+        assert "summary" in data
+        assert "timing_metrics" in data
+        assert "error_metrics" in data
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply tests/unit/test_core/test_metrics_export_lifecycle.py:51]: Addressed: `test_roundtrip_json` (lines 53-65 in test_metrics_export_lifecycle.py) asserts `panther_version` is a non-empty string, `error_count >= 1`, and `total_execution_time` exists. The version is dynamically imported from `panther.__version__`. Resolving.

File: panther/core/metrics/data_loader.py [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+
+        # Resource stats
+        resource = data.get("resource_metrics", {})
+        if isinstance(resource, dict):
+            cpu = resource.get("cpu_usage", {})
+            mem = resource.get("memory_usage", {})
+            if isinstance(cpu, dict) and cpu.get("average"):
+                summary["avg_cpu"] = f"{cpu['average']:.1f}%"
+                summary["peak_cpu"] = f"{cpu.get('peak', 0):.1f}%"
+            if isinstance(mem, dict) and mem.get("average"):
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply panther/core/metrics/data_loader.py]: Addressed: This is a duplicate of the already-resolved thread #1 (Sourcery comment on zero-valued averages). `data_loader.py:305` now uses `cpu.get("average") is not None` instead of truthiness checks. Resolving.

File: panther/cli_click/commands/run.py [hunk 8]
```diff
@@ -339,36 +339,42 @@ def run(
...
+                    metrics_output_path.mkdir(parents=True, exist_ok=True)
+                    exp_name = experiment_name or Path(config).stem
                     metrics_collector = MetricsCollector(
-                        interval=metrics_interval,
-                        output_dir=metrics_output_dir,
-                        export_format=metrics_export_format,
-                        disable_resource_monitoring=metrics_disable_resource_monitoring,
+                        experiment_name=exp_name,
+                        output_dir=metrics_output_path,
+                        collection_interval=float(metrics_interval),
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply panther/cli_click/commands/run.py:348]: Addressed: `--metrics-output-dir` help text updated to clarify: "Base directory for metrics collector state. Note: final exported metrics are written to <experiment_dir>/metrics/ during cleanup (default: outputs)". The MetricsCollector output_dir is for internal state; ExperimentManager.cleanup() controls the actual export path to experiment_dir/metrics/. Resolving.

File: panther/cli_click/commands/metrics.py [hunk 9]
```diff
@@ -61,60 +98,46 @@ def list(ctx, filter):
...
-                    click.echo(f"  • {metric['name']} ({metric['count']} samples)")
-                click.echo()
-
-            success_message("Metrics listing completed")
-        else:
-            warning_message("No metrics found matching the filter")
+    if filter_pattern:
+        import re
+
+        available = [m for m in available if re.search(filter_pattern, m["name"])]
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply panther/cli_click/commands/metrics.py]: Addressed: `metrics.py:109-113` now wraps the regex filter in `try/except re.error as e` with a user-friendly error message via `error_message(f"Invalid filter pattern ...")`. Resolving.

File: panther/core/observer/impl/metrics_observer.py [hunk 10]
```diff
@@ -46,8 +46,7 @@
 from panther.core.metrics.enums import MetricType, Phase
 from panther.core.observer.base.typed_observer_interface import ITypedObserver
 
-if TYPE_CHECKING:
-    from panther.core.metrics.resource_monitor import ResourceMonitor
+from panther.core.metrics.resource_monitor import ResourceMonitor
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply panther/core/observer/impl/metrics_observer.py]: Addressed: `metrics_observer.py:49-52` guards the import: `if PSUTIL_AVAILABLE: from ... import ResourceMonitor; else: ResourceMonitor = None`. This prevents ImportError when psutil is not installed. Resolving.

File: panther/core/metrics/data_loader.py [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+        raw = data.get("raw_metrics", {})
+        if isinstance(raw, dict):
+            for section_key in ("counters", "gauges", "histograms"):
+                section = raw.get(section_key, {})
+                if isinstance(section, dict):
+                    for name, value in section.items():
+                        count = (
+                            len(value)
+                            if isinstance(value, list)
+                            else (1 if value else 0)
```
[the remark this answers is not in the document]
Comment (ElNiak) [reply panther/core/metrics/data_loader.py]: Addressed: `data_loader.py:148` now uses `else (0 if value is None else 1)` - explicit None check instead of truthiness, so `value=0` correctly counts as 1 sample. Resolving.
